src/wellKnownDidConfiguration/index.ts

```diff
@@ -61,155 +51,269 @@ function checkOrigin(input: string) {
   }
 }
 
+export const DOMAIN_LINKAGE_CREDENTIAL_TYPE = 'DomainLinkageCredential' as const
+export const DATA_INTEGRITY_PROOF_TYPE = 'DataIntegrity' as const
+
+const suites = [ed25519.cryptosuite, sr25519.cryptosuite, es256k.cryptosuite]
+
 export async function createCredential(
-  signCallback: SignCallback,
+  signer: SignerInterface,
   origin: string,
-  didUri: DidUri
-): Promise<ICredentialPresentation> {
+  did: DidUri | ConformingDidDocument,
+  {
+    proofType = KILT_SELF_SIGNED_PROOF_TYPE,
+    expirationDate = new Date(Date.now() + 1000 * 60 * 60 * 24 * 365 * 5),
+  }: {
+    expirationDate?: Date
+    proofType?: typeof KILT_SELF_SIGNED_PROOF_TYPE | typeof DATA_INTEGRITY_PROOF_TYPE
+  } = {}
+): Promise<DomainLinkageCredential> {
   checkOrigin(origin)
 
-  const fullDid = await Did.resolve(didUri)
-
-  if (!fullDid?.document) {
-    throw new Error('No Did found: Please create a Full DID')
+  const document = await (async () => {
+    if (typeof did === 'string') {
+      const { didDocument } = await Did.resolveCompliant(did)
+      if (!didDocument) {
+        throw new Error('Failed to resolve DID ' + did)
+      }
+      return didDocument
+    } else if (typeof did === 'object' && did?.id) {
+      return did
+    } else {
+      throw new Error('must pass a DID or DID Document for the did parameter')
+    }
+  })()
+
+  if (!document.assertionMethod?.length) {
+    throw new Error('DID Document does not contain assertion key: Please add assertion key')
   }
 
-  const { document } = fullDid
-
-  const assertionKey = document.assertionMethod?.[0]
-
-  if (!assertionKey) {
-    throw new Error('Full DID doesnt have assertion key: Please add assertion key')
+  const credentialBody: Omit<DomainLinkageCredential, 'proof'> = {
+    '@context': [DEFAULT_VERIFIABLECREDENTIAL_CONTEXT, DID_CONFIGURATION_CONTEXT],
+    issuer: document.id,
+    issuanceDate: new Date().toISOString(),
+    expirationDate: expirationDate.toISOString(),
+    type: [DEFAULT_VERIFIABLECREDENTIAL_TYPE, DOMAIN_LINKAGE_CREDENTIAL_TYPE],
+    credentialSubject: {
+      id: document.id,
+      origin,
+    },
   }
 
-  const domainClaimContents = {
-    origin,
+  if (!signer) {
+    throw new Error('No signer available for an assertion method of the DID')
   }
 
-  const claim = Claim.fromCTypeAndClaimContents(ctypeDomainLinkage, domainClaimContents, document.uri)
+  switch (proofType) {
+    case KILT_SELF_SIGNED_PROOF_TYPE: {
+      const credential = {
+        ...credentialBody,
+        proof: {
+          type: KILT_SELF_SIGNED_PROOF_TYPE,
+          verificationMethod: signer.id,
+          proofPurpose: 'assertionMethod',
+        },
+      } as DomainLinkageCredential
+      const docHash = blake2AsU8a(JSON.stringify(credential))
+      const signature = await signer.sign({ data: docHash })
```

Comment:
Didn't we discover this proofs are basically useless or was that only for the socialKYC/

Reply:
they were, this is why I changed this. As you can see, I now sign the JSON representation of the entire credential, similar to how it would be done in a JWT. Because sporran currently doesn't care what you sign, it will still accept it - but this gives us an upgrade path, where sporran can stop accepting unsafe proofs at some point, and require that the signature is generated over credential contents.

Reply:
Ok, glad that you did change it.